src/sentry/api/endpoints/organization_teams.py

```diff
@@ -142,25 +143,42 @@ def post(self, request: Request, organization, **kwargs) -> Response:
 
         :pparam string organization_slug: the slug of the organization the
                                           team should be created for.
-        :param string name: the optional name of the team.
-        :param string slug: the optional slug for this team.  If
-                            not provided it will be auto generated from the
-                            name.
+        :qparam string name: the optional name of the team.
+        :qparam string slug: the optional slug for this team. If not provided it will be auto
+                             generated from the name.
+        :qparam bool set_team_admin: If this is true, the user is added to the as a Team Admin
+                                instead of regular member
         :auth: required
         """
         serializer = TeamPostSerializer(data=request.data)
 
-        if serializer.is_valid():
-            result = serializer.validated_data
+        if not serializer.is_valid():
+            raise ValidationError(serializer.errors)
 
+        result = serializer.validated_data
+        set_team_admin = result.get("set_team_admin")
+
+        if set_team_admin:
+            if not features.has("organizations:team-roles", organization) or not features.has(
+                "organizations:team-project-creation-all", organization
+            ):
+                raise ResourceDoesNotExist(
+                    detail="You do not have permission to join a new team as a team admin"
+                )
+            if not self.should_add_creator_to_team(request):
+                raise ValidationError(
+                    {"detail": "You do not have permission to join a new team as a Team Admin"}
+                )
+
+        # Wrap team creation and member addition in same transaction
+        with transaction.atomic():
             try:
-                with transaction.atomic():
-                    team = Team.objects.create(
-                        name=result.get("name") or result["slug"],
-                        slug=result.get("slug"),
-                        idp_provisioned=result.get("idp_provisioned", False),
-                        organization=organization,
-                    )
+                team = Team.objects.create(
+                    name=result.get("name") or result["slug"],
+                    slug=result.get("slug"),
+                    idp_provisioned=result.get("idp_provisioned", False),
+                    organization=organization,
+                )
             except IntegrityError:
```

Comment:
`transaction. Atomic` context manager checks if it left the block nicely, or through an exception. If it left through an exception then it will go through some cleanup operations including rollbacks.

See the callout in this section: https://docs.djangoproject.com/en/4.2/topics/db/transactions/#controlling-transactions-explicitly

If you're expanding the atomic block you'll need to move the except blocks.